### PR TITLE
add hack/bootstrap-fork-repos.sh script

### PR DESCRIPTION
**Pull request by mfojtik**
@deads2k this will help with initial fork repo bootstrap (and bumping versions).

We need to run:
```
ORIGIN_COMMIT="XXX" hack/bootstrap-fork-repos.sh
```

Where the `ORIGIN_COMMIT` points usually to a kube rebase commit or to a commit we want bot to start syncing the changes from.

The script will:

1) Create the branch if not exists (configurable in bash script)
2) Create empty commit with Origin-commit SHA
3) Push to repo

You need to run `go get -u gopkg.in/src-d/go-git.v4/...` before running bootstrap (need git lib).

(also I need somebody with push access to all those repos ;-)

### CODE REVIEW

File: tools/prepare-fork-repo/main.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	}
+	bs, err := remoteBranches(r.Storer)
+	if err != nil {
+		log.Fatalf("unable to list remote branches: %v", err)
+	}
+	var (
+		baseBranchHash plumbing.Hash
+		dstBranch      *plumbing.Reference
+	)
+	err = bs.ForEach(func(ref *plumbing.Reference) error {
```
Comment (sttts) [inline tools/prepare-fork-repo/main.go]: Why not directly lookup the refs you are interested in? Looping looks strange.

File: tools/prepare-fork-repo/main.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	return storer.NewReferenceFilteredIter(func(ref *plumbing.Reference) bool {
+		return ref.Name().IsRemote()
+	}, refs), nil
+}
+
+func main() {
```
Comment (sttts) [inline tools/prepare-fork-repo/main.go]: A paragraph of plain english describing what the tool does, would be nice

File: tools/prepare-fork-repo/main.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	if err := r.Storer.SetReference(dstBranch); err != nil {
+		log.Fatalf("unable to create %s branch: %v", *branchName, err)
+	}
+	if baseBranchHash.IsZero() {
+		log.Fatalf("base branch %s does not exists", baseBranchName)
+	}
+
+	w, _ := r.Worktree()
+
+	commit, err := w.Commit(fmt.Sprintf("Origin-commit: %s", *initialCommit), &git.CommitOptions{
```
Comment (sttts) [inline tools/prepare-fork-repo/main.go]: you don't even need worktree for this. Just create a commit object directly and store it.

**Comment by sttts** [comment]
Some nits. Otherwise lgtm.

File: tools/prepare-fork-repo/main.go (hunk 3 above)
Comment (mfojtik) [reply tools/prepare-fork-repo/main.go]: I tried but src-d/git complained that you cannot create commit on in-memory storage.

**Comment by mfojtik** [comment]
@sttts comments addressed

**Pull request closed without merging**